pkg/hintrunner/zero/zerohint_math.go

```diff
@@ -225,3 +225,149 @@ func createIsNNOutOfRangeHinter(resolver hintReferenceResolver) (hinter.Hinter,
+		Name: "IsPositive",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.math_utils import is_positive
+			//> ids.is_positive = 1 if is_positive(
+			//>     value=ids.value, prime=PRIME, rc_bound=range_check_builtin.bound) else 0
+
+			isPositiveAddr, err := dst.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			value, err := value.Resolve(vm)
+			if err != nil {
+				return err
+			}
+			valueFelt, err := value.FieldElement()
+			if err != nil {
+				return err
+			}
+
+			var v memory.MemoryValue
+			if utils.FeltIsPositive(valueFelt) {
+				v = memory.MemoryValueFromFieldElement(&utils.FeltOne)
+			} else {
+				v = memory.MemoryValueFromFieldElement(&utils.FeltZero)
+			}
+			return vm.Memory.WriteToAddress(&isPositiveAddr, &v)
+		},
+	}
+}
+
+func createIsPositiveHinter(resolver hintReferenceResolver) (hinter.Hinter, error) {
+	// This hint is executed for the negative values.
+	// If the value was non-negative, it's usually handled by the IsNN hint.
+
+	a, err := resolver.GetResOperander("a")
+	if err != nil {
+		return nil, err
+	}
+	return newIsNNOutOfRangeHint(a), nil
+}
+
+func newSplitIntAssertRangeHint(value hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "SplitIntAssertRange",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> assert ids.value == 0, 'split_int(): value is out of range.'
+
+			value, err := value.Resolve(vm)
+			if err != nil {
+				return err
+			}
+			valueFelt, err := value.FieldElement()
+			if err != nil {
+				return err
+			}
+			if valueFelt.IsZero() {
+				return fmt.Errorf("assertion `split_int(): value is out of range` failed")
+			}
+
+			return nil
+		},
+	}
+}
+
+func createSplitIntAssertRangeHinter(resolver hintReferenceResolver) (hinter.Hinter, error) {
+	value, err := resolver.GetResOperander("value")
+	if err != nil {
+		return nil, err
+	}
+	return newSplitIntAssertRangeHint(value), nil
+}
+
+func newSplitIntHint(output, value, base, bound hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "SplitIntHint",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> memory[ids.output] = res = (int(ids.value) % PRIME) % ids.base
+			//> assert res < ids.bound, f'split_int(): Limb {res} is out of range.'
+
+			outputAddr, err := output.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			base, err := base.Resolve(vm)
+			if err != nil {
+				return err
+			}
+			baseFelt, err := base.FieldElement()
+			if err != nil {
+				return err
+			}
+
+			value, err := value.Resolve(vm)
+			if err != nil {
+				return err
+			}
+			valueFelt, err := value.FieldElement()
+			if err != nil {
+				return err
+			}
+
+			bound, err := bound.Resolve(vm)
+			if err != nil {
+				return err
+			}
+			boundFelt, err := bound.FieldElement()
+			if err != nil {
+				return err
+			}
+
+			result := utils.FeltMod(valueFelt, baseFelt)
```

Comment:
Quick question: I saw you omitted this step `int(ids.value) % PRIME`, is it safe to do so?
I understand what is stored in `value` should fit in a felt, in that case it wouldn't be necessary to do it. Should we add a comment clarifying this?

Reply:
I commented on it in one of the places and decided that it could be excessive here.
All operations on felts we use are module prime - it's stated in the felt comments.